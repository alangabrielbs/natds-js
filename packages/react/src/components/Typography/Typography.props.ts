export interface TypographyProps {
  /**
   * Specify an optional className to be added to your Typography
   */
  className?: string

  /**
   * Optional ID for testing
   */
  testID?: string;
}
