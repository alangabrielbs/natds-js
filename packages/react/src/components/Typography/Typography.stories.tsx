import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Typography, TypographyProps } from '.'

const componentStatus = `
- - -

**NOTE FOR UXs**: This component is available in the following variants:

  - ❌ Standard

With the following attribute statuses:

  - **Attributes:**
    - ❌ \`Types\`

- - -
`

export default {
  title: 'Components/Typography',
  component: Typography,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
} as Meta

export const Playground: Story<TypographyProps> = (args) => (
  <Typography {...args} />
)
