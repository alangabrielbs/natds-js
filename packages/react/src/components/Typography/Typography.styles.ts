/* eslint-disable complexity */
import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { TypographyProps } from './Typography.props'

type TypographyVariant = Pick<TypographyProps, 'variant'>

const setStyleByVariant = (theme: Theme) => ({ variant }: TypographyVariant) => {
  const headingBase = {
    fontFamily: theme.typography.fontFamily
  }

  // const sutitleBase = {
  //   fontFamily: theme.typography.fontSize.
  // }

  switch (variant) {
    case 'heading1':
      return { ...headingBase, fontSize: 96 }
    case 'heading2':
      return { ...headingBase, fontSize: 60 }
    case 'heading3':
      return { ...headingBase, fontSize: 48 }
    case 'heading4':
      return { ...headingBase, fontSize: 34 }
    case 'heading5':
      return { ...headingBase, fontSize: 24 }
    case 'heading6':
      return { ...headingBase, fontSize: 20 }
    case 'subtitle1':
      return { ...headingBase, fontSize: 20 }
    default:
      break
  }
}

const styles = createUseStyles((theme: Theme) => ({
  example: setStyleByVariant(theme)

}))

export default styles
