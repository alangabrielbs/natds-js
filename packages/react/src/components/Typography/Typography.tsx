import React, { useEffect, useState } from 'react'
import styles from './Typography.styles'
import { TypographyProps } from './Typography.props'

const Typography = React.forwardRef<HTMLElement, TypographyProps>(
  (props, ref) => {
    const {
      variant = 'body1',
      ...rest
    } = props
    const { example } = styles({ variant })
    const [CustomTag, setCustomTag] = useState('p' as keyof JSX.IntrinsicElements)

    useEffect(() => {
      const suffix = variant[variant.length - 1]
      let nameElement

      if (variant.includes('heading')) {
        nameElement = `h${suffix}`
      } else if (variant === 'caption' || variant === 'overline') {
        nameElement = 'span'
      } else {
        nameElement = 'p'
      }

      setCustomTag(nameElement as keyof JSX.IntrinsicElements)
    }, [variant])

    return <CustomTag className={example}>hello</CustomTag>
  }
)

export default Typography
