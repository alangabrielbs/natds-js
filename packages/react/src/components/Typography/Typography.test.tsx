import React from 'react'
import Typography from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('Typography component', () => {
  it('should render the component as body1 when variant is not defined', () => {
    const { styles, component } = renderWithTheme(<Typography>hello</Typography>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
    expect(component.getByText(/hello/i)).toBeInTheDocument()
    expect(styles.includes('font-size: 16px;')).toBeTruthy()
  })
})
