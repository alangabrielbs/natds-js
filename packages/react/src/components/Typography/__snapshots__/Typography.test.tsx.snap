// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component should render the component as body1 when variant is not defined 1`] = `
Array [
  ".text-0-2-1 {}
.text-d0-0-2-2 {
  font-family: Helvetica Now Display;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5;
  letter-spacing: 0.51px;
  text-transform: none;
}",
  <div>
    <div>
      <p
        class="text-0-2-1 text-d0-0-2-2"
      >
        hello
      </p>
    </div>
  </div>,
]
`;
